Components/ProductToShop.php

```diff
@@ -186,12 +186,26 @@ public function commit()
      */
     public function insertOrUpdate(Product $product)
     {
+        $product = $this->eventManager->filter(
+            'Connect_ProductToShop_InsertOrUpdate_Before',
+            $product
+        );
+
         // todo@dn: Set dummy values and make product inactive
         if (empty($product->title) || empty($product->vendor)) {
             return;
         }
 
         $detail = $this->helper->getArticleDetailModelByProduct($product);
+        $detail = $this->eventManager->filter(
+            'Connect_Merchant_Get_Article_Detail_After',
+            $detail,
```

Comment:
```$detail``` can be null when the product doesn't exist in your DB. So I think this event must be moved on line 247, before ```if (!empty($product->sku)) {```. Then ```$detail``` object always exists. In some cases it will be completely empty, but exists. Otherwise in you classes you have to check ``` if (!$detail) { return; } else { //do something }.

@sans0r what do you think ?